widgets/SalesforceBatchSource-batchsource.json

```diff
@@ -0,0 +1,89 @@
+{
+  "metadata": {
+    "spec-version": "1.5"
+  },
+  "configuration-groups": [
+    {
+      "label": "Reference",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Reference Name",
+          "name": "referenceName"
+        }
+      ]
+    },
+    {
+      "label": "Authentication",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Username",
+          "name": "username"
+        },
+        {
+          "widget-type": "password",
+          "label": "Password",
+          "name": "password"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Client Id",
+          "name": "clientId"
+        },
+        {
+          "widget-type": "password",
+          "label": "Client Secret",
+          "name": "clientSecret"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Login Url",
+          "name": "loginUrl",
+          "widget-attributes" : {
+            "default": "https://login.salesforce.com/services/oauth2/token"
+          }
+        }
+      ]
+    },
+    {
+      "label": "Query",
+      "properties": [
+        {
+          "widget-type": "textarea",
+          "label": "SOQL Query",
+          "name": "query",
+          "widget-attributes" : {
+            "placeholder": "SELECT <offset field>, ... FROM <object> WHERE <offset field> > <offset> ORDER BY <offset field>"
+          },
+          "plugin-function": {
+            "method": "POST",
+            "widget": "getSchema",
+            "output-property": "schema",
+            "plugin-method": "getSchema"
+          }
+        }
+      ]
+    }
+  ],
+  "outputs": [
+    {
+      "name": "schema",
+      "label": "schema",
+      "widget-type": "schema",
+      "widget-attributes": {
+        "schema-types": [
+          "boolean",
```

Comment:
Doesn't support all the types that are handled above. Could we make sure we handle all types supported by SF here.

Reply:
Here are all types salesforce supports:

> string("string"), picklist("picklist"), multipicklist("multipicklist"), combobox("combobox"), reference("reference"), base64("base64"), _boolean("boolean"), currency("currency"), textarea("textarea"), _int("int"), _double("double"), percent("percent"), phone("phone"), id("id"), date("date"), datetime("datetime"), time("time"), url("url"), email("email"), encryptedstring("encryptedstring"), datacategorygroupreference("datacategorygroupreference"), location("location"), address("address"), anyType("anyType"), json("json"), complexvalue("complexvalue"), _long("long");

we currently support all types based on integer (and long), float (and double), string, boolean, date, timestamp. 
Regarding picklist and multipicklist (in sf they are enums like enums in java) and we represent them as strings.

As far as I see we support all the types saleforce supports.